app/services/DynamoArchivedChannelTests.scala

```diff
@@ -0,0 +1,64 @@
+package services
+
+import com.typesafe.scalalogging.StrictLogging
+import models.DynamoErrors._
+import models._
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+import software.amazon.awssdk.services.dynamodb.model._
+import zio.blocking.effectBlocking
+import zio.duration.durationInt
+import zio.stream.ZStream
+import zio.{ZEnv, ZIO}
+
+import scala.jdk.CollectionConverters._
+
+
+class DynamoArchivedChannelTests(stage: String, client: DynamoDbClient) extends StrictLogging {
+
+  private val tableName = s"support-admin-console-archived-channel-tests-$stage"
+
+  // Sends a batch of write requests, and returns any unprocessed items
+  private def putAll(writeRequests: List[WriteRequest]): ZIO[ZEnv, DynamoPutError, List[WriteRequest]] =
+    effectBlocking {
+      val batchWriteRequest =
+        BatchWriteItemRequest
+          .builder
+          .requestItems(Map(tableName -> writeRequests.asJava).asJava)
+          .returnConsumedCapacity(ReturnConsumedCapacity.TOTAL)
+          .build()
+
+      val result = client.batchWriteItem(batchWriteRequest)
+      logger.info(s"BatchWriteItemResponse: $result")
+
+      result.unprocessedItems().asScala
```

Comment:
Just out of interest - why might there be `unprocessedItems` here if `putAllBatched` (which calls this) is splitting into batches of 25 alread?

Reply:
Good question - it's in case of errors. From the docs:
>>  If any requested operations fail because the table's provisioned throughput is exceeded or an internal processing failure occurs, the failed operations are returned in the UnprocessedItems response parameter.
https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchWriteItem.html